### PR TITLE
improve docs for machine learning workspace

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

File: website/docs/r/machine_learning_workspace.html.markdown [hunk 1]
```diff
@@ -167,8 +169,6 @@ The following arguments are supported:
 
 * `key_vault_id` - (Required) The ID of key vault associated with this Machine Learning Workspace. Changing this forces a new resource to be created.
 
--> NOTE: Azure Machine Learning Workspace will turn on purge protection on the key vault if it is not enabled.
```
Comment (ms-henglu) [inline website/docs/r/machine_learning_workspace.html.markdown:170]: From service team
> This behavior does not longer hold. We do not longer turn on purge protection. Azure Machine Learning validates whether soft delete and purge protection are enabled when bringing customer-managed encryption resources. CMK workspace creation fails if soft delete and purge protection are not enabled.

**Review by denniseik — APPROVED** [review]
Thanks!

File: website/docs/r/machine_learning_workspace.html.markdown [hunk 2]
```diff
@@ -185,21 +185,19 @@ The following arguments are supported:
 
 * `description` - (Optional) The description of this Machine Learning Workspace.
 
-* `discovery_url` - (Optional) The URL for the discovery service to identify regional endpoints for machine learning experimentation services.
```
Comment (katbyte) [inline website/docs/r/machine_learning_workspace.html.markdown:188]: how come we are removing this? it is still in schema?

Comment (ms-henglu) [reply website/docs/r/machine_learning_workspace.html.markdown:188]: It's a computed property and it's already in `Attributes Reference` section.

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu, LGTM 👍🏼

**Pull request merged**